scripts/github-action-permit-generator.ts

```diff
@@ -0,0 +1,116 @@
+import * as github from "@actions/github";
+import { Octokit } from "@octokit/rest";
+import { createClient } from "@supabase/supabase-js";
+import { createAdapters } from "../src/adapters";
+import { Database } from "../src/adapters/supabase/types/database";
+import { generatePayoutPermit } from "../src/handlers";
+import { Context } from "../src/types/context";
+import { PermitGenerationSettings, PermitRequest } from "../src/types/plugin-input";
+import { Value } from "@sinclair/typebox/value";
+import { envSchema } from "../src/types/env";
+import * as fs from "fs";
+
+/**
+ * Generates all the permits based on the current github workflow dispatch.
+ */
+export async function generatePermitsFromGithubWorkflowDispatch() {
+  const runId = github.context.runId;
+
+  // These are necessary to ensure the type checks and tests pass.
+  process.env["NFT_MINTER_PRIVATE_KEY"] = "";
+  process.env["NFT_CONTRACT_ADDRESS"] = "";
```

Comment:
We should fix the type checks in another pull and make these optional probably? @whilefoo rfc

Reply:
yeah this should be done with Typebox

Reply:
Is this PR completed?and which ISSUE association code do I use?